Docker/JobDefinition.php

```diff
@@ -0,0 +1,218 @@
+<?php
+namespace Keboola\DockerBundle\Docker;
+
+use Keboola\Syrup\Exception\UserException;
+use Symfony\Component\Config\Definition\Exception\InvalidConfigurationException;
+
+class JobDefinition
+{
+    /**
+     * @var
+     */
+    private $configId;
+
+    /**
+     * @var string
+     */
+    private $rowId;
+
+    /**
+     * @var string
+     */
+    private $configVersion;
+
+    /**
+     * @var string
+     */
+    private $rowVersion;
+
+    /**
+     * @var Component
+     */
+    private $component;
+
+    /**
+     * @var array
+     */
+    private $configuration = [];
+
+    /**
+     * @var array
+     */
+    private $state = [];
+
+    /**
+     * @var bool
+     */
+    private $isDisabled = false;
+
+    public function __construct($configuration)
+    {
+        $this->configuration = $this->normalizeConfiguration($configuration);
```

Comment:
fakt si myslim, ze by to melo byt uplne immutable, ty settery se nepouzivaji nikde jinde nez pri vytvoreni

Reply:
viz https://github.com/keboola/docker-bundle/pull/196#discussion_r144222886